GVFS/GVFS/CommandLine/DiagnoseVerb.cs

```diff
@@ -390,7 +396,8 @@ private void RecursiveFileCopyImpl(string sourcePath, string targetPath, bool co
                 try
                 {
                     string fileExtension = Path.GetExtension(fileName);
-                    if (!string.Equals(fileExtension, ".exe", StringComparison.OrdinalIgnoreCase))
+                    if (!string.Equals(fileExtension, WindowsExecutableExtension, StringComparison.OrdinalIgnoreCase) &&
+                        !string.Equals(fileName, MacNamedPipeSocketName, StringComparison.OrdinalIgnoreCase))

```

Comment:
Maybe the platform should have a method like `bool ShouldCopyForDiagnose(string filename)`?

Reply:
Or `IsExecutable`

Reply:
> Maybe the platform should have a method like bool ShouldCopyForDiagnose(string filename)

I think this would be cleaner, but I'm not sure if it violates the spirit of the platforms.  @sanoursa, any preferences\objections?

I actually had started down the road of adding `IsExecutable` and `IsSocket` to the platform (which seemed like a better fit), but it ended up being quite complicated on Mac (and didn't seem like something we absolutely needed for `gvfs diagnose`).  The trouble I ran into was that the output `struct` of `stat(...)` can vary based on `ifdefs`, and it wasn't clear how we could determine which format would be returned when PInvoking to libc.

Reply:
Yep as we discussed offline, the `GVFSPlatform` is intended to represent behaviors of the platform, not GVFS-specific business logic. `IsExecutable` and `IsSocket` are a better fit than `ShouldCopyForDiagnose`.

Reply:
I'll go ahead and add `IsExecutable` and `IsSocket`

Reply:
> but it ended up being quite complicated on Mac 

I figured out the issue I had here as well, I will bring back my earlier changes for PInvoking to `stat`